src/node.rs

```diff
@@ -557,6 +559,23 @@ impl<S: std::fmt::Debug + ForkSource> InMemoryNodeInner<S> {
             Some(revert) => Err(revert.revert_reason),
         }
     }
+
+    /// Sets the `impersonated_account` field of the node.
+    /// This field is used to override the `tx.initiator_account` field of the transaction in the `run_l2_tx` method.
+    pub fn set_impersonated_account(&mut self, address: Address) {
+        self.impersonated_account = Some(address);
+
+        // Use SystemContracts without signature verification
+        self.system_contracts = SystemContracts::from_options(&Options::BuiltInWithoutSecurity);
```

Comment:
Maybe we should set no security SC only for the transaction from impersonated account, not for the whole state, while we are impersonating?

Reply:
Agreed. If we implement the other comment/suggestion of only impersonating when `tx.from` is in the impersonated accounts list, this would cause some transactions to be run without security checks unexpectedly.

As noted here, we should probably move the logic of setting `self.system_contracts` down to either `run_l2_tx` or `send_raw_transaction`. I personally lean towards `run_l2_tx` in case we implement `eth_sendTransaction`, but up to you.